### ISSUE TITLE
[Bug]: term filtering will be failed on indexed scalar field

### ISSUE THREAD

**Issue by longjiquan** [milvus-io/milvus#20835]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

https://github.com/milvus-io/milvus/blob/45205da8f184f908f9cbf222f8ee3e0edb26cf72/internal/core/src/query/visitors/ExecExprVisitor.cpp#L754-L816

**Comment by longjiquan** [comment]
This is the root cause of https://github.com/milvus-io/milvus/issues/20534

**Comment by longjiquan** [comment]
Fixed in #20840 and #20882, so close this.
